authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
++
+[source,json]
+----
+{
+	"issuer": "$OPENID_BUCKET_URL",
+	"jwks_uri": "${OPENID_BUCKET_URL}/keys.json",
+    "response_types_supported": [
+        "id_token"
+    ],
+    "subject_types_supported": [
+        "public"
+    ],
+    "id_token_signing_alg_values_supported": [
+        "RS256"
+    ],
+    "claims_supported": [
+        "aud",
+        "exp",
+        "sub",
+        "iat",
+        "iss",
+        "sub"
+    ]
+}
+----
+
+. Upload the OIDC configuration:
++
+[source,terminal]
+----
+$ aws s3api put-object --bucket <oidc-bucket-name> --key keys.json --body ./keys.json
+----
++
+[source,terminal]
+----
+$ aws s3api put-object --bucket <oidc-bucket-name> --key '.well-known/openid-configuration' --body ./openid-configuration
+----
+
+. Allow the AWS IAM OIDC identity provider to read these files:
++
+[source,terminal]
+----
+$ aws s3api put-object-acl --bucket <oidc-bucket-name> --key keys.json --acl public-read
+----
++
+[source,terminal]
+----
+$ aws s3api put-object-acl --bucket <oidc-bucket-name> --key '.well-known/openid-configuration' --acl public-read
+----
+
+. Create an AWS IAM OIDC identity provider:
+
+.. Get the certificate chain from the server that hosts the OIDC configuration:
++
+[source,terminal]
+----
+$ echo | openssl s_client -servername s3.$<aws-region>.amazonaws.com -connect s3.$<aws-region>.amazonaws.com:443 -showcerts 2>/dev/null | awk '/BEGIN/,/END/{ if(/BEGIN/){a++}; out="cert"a".pem"; print >out}'
+----
+
+.. Calculate the fingerprint for the certificate at the root of the chain:
++
+[source,terminal]
+----
+$ export BUCKET_FINGERPRINT=$(openssl x509 -in cert<number>.pem -fingerprint -noout | sed -e 's/.*Fingerprint=//' -e 's/://g')
+----
++
+Where `<number>` is the highest number in the files that were saved. For example, if two is the highest number in the files that were saved, use `cert2.pem`.
+
+.. Create the identity provider:
++
+[source,terminal]
+----
+$ aws iam create-open-id-connect-provider --url $OPENID_BUCKET_URL --thumbprint-list $BUCKET_FINGERPRINT
+----
+
+.. Save the returned ARN of the newly created identity provider to `$AWS_IAM_OPENID_ARN`.
+
+. Generate IAM roles:
+
+.. Locate all `CredentialsRequest` CRs in this release image that target the cloud you are deploying on:
++
+[source,terminal]
+----
+$ oc adm release extract quay.io/openshift-release-dev/ocp-release:4.y.z-x86_64 --credentials-requests --cloud=aws
+----
+
+.. For each `CredentialsRequest` CR, create an IAM role that grants the necessary permissions and establishes a trust relationship that trusts the  identity provider previously created.
++
+For example, for the openshift-machine-api-operator `CredentialsRequest` CR in `0000_30_machine-api-operator_00_credentials-request.yaml`, create an IAM role that allows an identity from the created OIDC provider created for the cluster:
++
+[source,terminal]
+----
+$ aws iam get-role --role-name openshift-machine-api-aws-cloud-credentials
```

Comment:
Does this command create the role or just display it? See: https://docs.aws.amazon.com/cli/latest/reference/iam/get-role.html

Reply:
@joelddiaz  similarly here, should we tell them to [create-role](https://docs.aws.amazon.com/cli/latest/reference/iam/create-role.html) and to refer to the example for what they need to put into it?

Reply:
Example output of what the role should look like once the role is created/modified.